### PR TITLE
Add telemetry mem_spike test

### PR DESCRIPTION
**Pull request by zbud-msft**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?

Add integration test for verifying that telemetry server can handle client creating many connections

#### How did you do it?

Run python gnmi client to create large number of connections. If telemetry server has fix, it will pass test and will not fail memory utilization check

#### How did you verify/test it?

Run test case

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/telemetry/telemetry_utils.py:115:20: F821 undefined name 'wait_until'<br/>tests/telemetry/test_telemetry.py:59:1: E302 expected 2 blank lines, found 1<br/>tests/telemetry/test_telemetry.py:76:1: E302 expected 2 blank lines, found 1<br/>tests/telemetry/test_telemetry.py:94:1: E302 expected 2 blank lines, found 1<br/>tests/telemetry/test_telemetry.py:108:1: E302 expected 2 blank lines, found 1<br/>tests/telemetry/test_telemetry.py:118:20: E225 missing whitespace around operator<br/>tests/telemetry/test_telemetry.py:153:1: E302 expected 2 blank lines, found 1<br/>tests/telemetry/test_telemetry.py:162:106: E251 unexpected spaces around keyword / parameter equals<br/>tests/telemetry/test_telemetry.py:162:108: E251 unexpected spaces around keyword / parameter equals<br/>tests/telemetry/test_telemetry.py:169:76: E261 at least two spaces before inline comment<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/telemetry/test_telemetry.py [hunk 1]
```diff
@@ -143,10 +164,41 @@ def test_virtualdb_table_streaming(duthosts, enum_rand_one_per_hwsku_hostname, p
...
+    client_thread.start()
+
+    for i in range(MEMORY_CHECKER_CYCLES):
+        ret = duthost.shell("python3 /usr/bin/memory_checker telemetry 419430400", module_ignore_errors=True)
+        assert ret["rc"] == 0, "Memory utilization has exceeded threshold"
+        time.sleep(MEMORY_CHECKER_WAIT)
+
+    client_thread.join()
+
+    postcheck_critical_processes(duthost, "TELEMETRY")
```
Comment (ganglyu) [inline tests/telemetry/test_telemetry.py:204]: Why do you check critical process here? Telemetry container restarted for some reasons?

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by rhsmci** [comment]
@zbud-msft 
This test failed with 202111 branch of sonic-buildimage.
As the Description mentioned: 
"Run python gnmi client to create large number of connections. If telemetry server has fix, it will pass test and will not fail memory utilization check"
Could you identify the fix for telemetry server? Thank you.

**Comment by rhsmci** [comment]
> > @zbud-msft This test failed with 202111 branch of sonic-buildimage. As the Description mentioned: "Run python gnmi client to create large number of connections. If telemetry server has fix, it will pass test and will not fail memory utilization check" Could you identify the fix for telemetry server? Thank you.
> 
> Fix is not in 202111 which is why this test failed.

@zbud-msft Is it the fix? https://github.com/sonic-net/sonic-gnmi/commit/f576fe011aa49805b0ba38c9f6bc3697cfb34409

**Comment by zbud-msft** [comment]
> > > @zbud-msft This test failed with 202111 branch of sonic-buildimage. As the Description mentioned: "Run python gnmi client to create large number of connections. If telemetry server has fix, it will pass test and will not fail memory utilization check" Could you identify the fix for telemetry server? Thank you.
> > 
> > 
> > Fix is not in 202111 which is why this test failed.
> 
> @zbud-msft Is it the fix? [sonic-net/sonic-gnmi@f576fe0](https://github.com/sonic-net/sonic-gnmi/commit/f576fe011aa49805b0ba38c9f6bc3697cfb34409)

https://github.com/sonic-net/sonic-gnmi/pull/99, has fix

**Pull request merged**